kong/api/routes/config.lua

```diff
@@ -89,6 +323,16 @@ return {
         if check_hash and err_t and err_t.error == "configuration is identical" then
           return kong.response.exit(304)
         end
+
+        if err_t["@input"] then
```

Comment:
Feedback wanted re: this approach.

Augmenting the existing schema/declarative code to support a different "shape" of error table is a _very_ big and intrusive task, so my first attempt makes a small change to the declarative code (attaching the processed user input) and handles all of the parsing after-the-fact.

Is this a sane approach, or do we need to rethink it? And if we are okay with this approach, are there any suggestions for improvement?

Reply:
update: this code was moved into `kong.db.errors:declarative_config_flattened()`, which is IMO an improvement upon it living in the API code.